source/Sylvan.Data.Csv/CsvDataReader.cs

```diff
@@ -84,14 +84,28 @@ enum State
 		// the state when the end of the record set is reached
 		// not necessarily the end of the file when in multi-result set mode.
 		End,
-		// the state when the reader has bee closed/disposed.
+		// the state when the reader has been closed/disposed.
 		Closed,
+		// the reader has previously thrown an exception
+		Error,
 	}
 
-	void ValidateState()
+	void ValidateState(int ordinal)
 	{
 		if (this.state != State.Open)
 		{
+			if (this.state == State.Error)
+			{
+				if (ordinal < this.curFieldCount - 1)
+				{
+					return;// can read the fields up to the one that had the error.
+				}
+				else
+				{
+					// TODO:
+					throw new InvalidOperationException("The current row bla blah TODO");
```

Comment:
Why not throw pendingException here instead, which contains more information?

Reply:
Nevermind. I think you're right about this one. I was thinking about the other InvalidOperationException that is thrown from subsequent Read calls.